lib/remoteproc/elf_loader.c

```diff
@@ -445,51 +445,49 @@ int elf_load_header(const void *img_data, size_t offset, size_t len,
 		    offset + len < phdrs_offset + phdrs_size) {
 			*noffset = phdrs_offset;
 			*nlen = phdrs_size;
-			return (int)*load_state;
+			return *load_state;
 		}
-		/* caculate the programs headers offset to the image_data */
+		/* calculate the programs headers offset to the image_data */
 		phdrs_offset -= offset;
-		img_phdrs = (const void *)
-			    ((const char *)img_data + phdrs_offset);
-		phdrs = (char **)elf_phtable_ptr(*img_info);
-		(*phdrs) = metal_allocate_memory(phdrs_size);
+		img_phdrs = (const char *)img_data + phdrs_offset;
+		phdrs = elf_phtable_ptr(*img_info);
+		*phdrs = metal_allocate_memory(phdrs_size);
 		if (*phdrs == NULL)
 			return -RPROC_ENOMEM;
-		memcpy((void *)(*phdrs), img_phdrs, phdrs_size);
+		memcpy(*phdrs, img_phdrs, phdrs_size);
 		*load_state = ELF_STATE_WAIT_FOR_SHDRS |
 			       RPROC_LOADER_READY_TO_LOAD;
 	}
 	/* Get ELF Section Headers */
 	if ((*load_state & ELF_STATE_WAIT_FOR_SHDRS) != 0) {
 		size_t shdrs_size;
 		size_t shdrs_offset;
-		char **shdrs;
+		void **shdrs;
```

Comment:
void *shdrs;

Reply:
> void *shdrs;

Can't change to void *, since we need save the allocated memory:
*shdrs = metal_allocate_memory(shdrs_size);